src/components/opt-in-out/index.tsx

```diff
@@ -76,29 +53,12 @@ export default function OptInOut({ platform, redirectPath }: OptInOutProps) {
     if (typeof optInFrom == 'string' && isOptInPlatformOption(optInFrom)) {
       makeBetaWelcomeToast(optInFrom)
     }
-  }, [optInFrom, handleOptOut])
+  }, [optInFrom])
 
   // Return early if not opted in
   if (optedIn !== 'true') {
     return null
   }
 
-  return (
-    <div>
-      <Button
-        color="tertiary"
-        text="Leave Beta"
-        icon={<IconSignOut16 />}
-        iconPosition="trailing"
-        onClick={openDialog}
-      />
-      <Dialog onDismiss={closeDialog} isOpen={showDialog} label="Opt out form">
-        <OptOutForm
-          onSubmit={handleOptOut}
-          onDismiss={closeDialog}
-          platform="learn"
-        />
-      </Dialog>
-    </div>
-  )
+  return <OptOutButtonAndDialog handleOptOut={handleOptOut} />
```

Comment:
nit: Personal preference really, but not sure if the abstraction here is really valuable. There's no other UI rendered and the file here is fairly small.